.github/workflows/branch-protection.yml

```
@@ -0,0 +1,61 @@
+      - "scripts/branch-protection*"
+  schedule:
+    # Weekly, so drift introduced through the UI surfaces on its own rather than
+    # waiting for someone to touch the ruleset file.
+    - cron: "0 12 * * 1"
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  # Required to read rulesets. If the token is not granted it, the script exits
+  # 3 and this job warns instead of failing.
+  administration: read
```

actionlint (1.7.12)

[error] unknown permission scope "administration". all available permission scopes are "actions", "artifact-metadata", "attestations", "checks", "contents", "deployments", "discussions", "id-token", "issues", "models", "packages", "pages", "pull-requests", "repository-projects", "security-events", "statuses"

(permissions)